### PR TITLE
[qa] Handle zombies and cluttered tmpdirs

### PR DESCRIPTION
**Pull request by maflcko**
The first commit will make the tests fail and print a warning when the tmpdir exists.

The second commit will make the tests skip over any existing zombie daemons with high probability.

Both commit should improve the bad user experience reported in #9086. If the second commit is too controversial, I am happy to drop.

### CODE REVIEW

File: qa/pull-tester/rpc-tests.py [hunk 1]
```diff
@@ -246,14 +246,18 @@ def __init__(self, num_tests_parallel, test_list=None, flags=None):
         self.test_list = test_list
         self.flags = flags
         self.num_running = 0
+        # In case there is a graveyard of zombie bitcoinds, we can apply a
+        # psoudorandom offset to hopefully jump over them.
```
Comment (laanwj) [inline qa/pull-tester/rpc-tests.py]: pseudorandom :)

**Comment by laanwj** [comment]
utACK, hopefully improves the port-allocation issue in practice without introducing random collisions.

**Comment by MarcoFalke** [comment]
When tests are run in parallel through rpc-tests.py the ports must never collide, otherwise it is a bug. (Running several rpc-tests.py is not supported right now, though)

The second commit should only have an effect when there is at least one zombie instance, which still listens on a port and prevents subsequent tests from executing.

**Comment by laanwj** [comment]
> When tests are run in parallel through rpc-tests.py the ports must never collide, otherwise it is a bug. 

Agreed. 
(I guess there's the case of collisions with random other software listening on ports, but avoiding this is extremely difficult and this change doesn't make it statistically worse)

**Pull request merged**